backend/src/main/kotlin/blue/mild/covid/vaxx/service/VaccinationSlotService.kt

```diff
@@ -0,0 +1,127 @@
+        val createdIds = mutableListOf<EntityId>()
+        while (ts.plusMillis(createDto.durationMillis).isBefore(createDto.to.plusMillis(1))) {
+            val to = ts.plusMillis(createDto.durationMillis)
+            for (queue in 0 until createDto.bandwidth) {
+                createdIds.add(
+                    vaccinationSlotRepository.addVaccinationSlot(
+                        locationId = location.id,
+                        patientId = null,
+                        queue = queue,
+                        from = ts,
+                        to = to,
+                    )
+                )
+            }
+            ts = to
+        }
+
+        return createdIds
+    }
+
+    /**
+     * Filters the database with the conjunction (and clause) of the given properties.
+     */
+    suspend fun getSlotsByConjunctionOf(
+        id: EntityId? = null,
+        locationId: EntityId? = null,
+        fromMillis: Long? = null,
+        toMillis: Long? = null,
+        status: VaccinationSlotStatus? = DEFAULT_STATUS,
+    ): List<VaccinationSlotDtoOut> {
+        // To Consider: Maybe when id is specified all the other parameter should be
+        val fromI = Instant.ofEpochMilli(fromMillis ?: Instant.EPOCH.toEpochMilli())
+        // MartinLLama: With Instant.MAX it was failing on Exception occurred in the application: long overflow
+        val toI = Instant.ofEpochMilli(toMillis ?: MAX_FUTURE.toEpochMilli())
+        val usedStatus = status ?: if (id == null) DEFAULT_STATUS else VaccinationSlotStatus.ALL
+
+        return vaccinationSlotRepository.get {
+            Op.TRUE
+                .andWithIfNotEmpty(id, VaccinationSlots.id)
+                .andWithIfNotEmpty(locationId, VaccinationSlots.locationId)
+                .and { VaccinationSlots.from.greaterEq(fromI) }
+                .and { VaccinationSlots.to.lessEq(toI) }
+                .and { when(usedStatus) {
+                    // MartinLLama: I do not know how to write ALL as match all = 1 == 1
+                    VaccinationSlotStatus.ALL -> {VaccinationSlots.id.isNotNull()}
+                    VaccinationSlotStatus.ONLY_FREE -> VaccinationSlots.patientId.isNull()
+                    VaccinationSlotStatus.ONLY_OCCUPIED -> VaccinationSlots.patientId.isNotNull()
+                }}
+        }
+    }
+
+    @Suppress("TooGenericExceptionThrown", "LongParameterList")
+    suspend fun updateSlot(
+        id: EntityId? = null,
+        locationId: EntityId? = null,
+        fromMillis: Long? = null,
+        toMillis: Long? = null,
+        status: VaccinationSlotStatus?,
+        patientId: EntityId?,
+    ): VaccinationSlotDtoOut {
+        val availableSlots = getSlotsByConjunctionOf(
+            id = id,
+            locationId = locationId,
+            fromMillis = fromMillis,
+            toMillis = toMillis,
+            status = status,
+        )
+
+        if (availableSlots.isEmpty()) {
+            throw Exception("There are no available slots")
```

Comment:
No `Exception` please, something like not found again.

Reply:
I do not know how do you want to solve error signalization. This will happen if the UI either messes up parameters or when there is last free slot and multiple people in the same time wants to allocate it.

There is more problems in this part of the code. If `slotId` is used => it just rewrites the original person => original person looses its slot => this information is not propagated anywhere.
So there should be an option to use `slotId` to be able to clear specific slot, however there is a risk of overrides.